From 070c2faefd3efc25062ddcceadeb6989920f18e0 Mon Sep 17 00:00:00 2001
From: Matthew Nelson <developer@matthewnelson.io>
Date: Fri, 13 Oct 2023 07:09:56 -0400
Subject: [PATCH 03/11] Remove JDBC extensions

---
 Makefile | 12 +-----------
 1 file changed, 1 insertion(+), 11 deletions(-)

diff --git a/Makefile b/Makefile
index 53635d2..4de7e05 100644
--- a/Makefile
+++ b/Makefile
@@ -69,18 +69,8 @@ clean: clean-native clean-java clean-tests
 
 $(SQLITE_OUT)/sqlite3.o : $(SQLITE_UNPACKED)
 	@mkdir -p $(@D)
-	perl -p -e "s/sqlite3_api;/sqlite3_api = 0;/g" \
-	    $(SQLITE_SOURCE)/sqlite3ext.h > $(SQLITE_OUT)/sqlite3ext.h
-# insert a code for loading extension functions
-	perl -p -e "s/^opendb_out:/  if(!db->mallocFailed && rc==SQLITE_OK){ rc = RegisterExtensionFunctions(db); }\nopendb_out:/;" \
-	    $(SQLITE_SOURCE)/sqlite3.c > $(SQLITE_OUT)/sqlite3.c.tmp
-# register compile option 'JDBC_EXTENSIONS'
-# limits defined here: https://www.sqlite.org/limits.html
-	perl -p -e "s/^(static const char \* const sqlite3azCompileOpt.+)$$/\1\n\n\/* This has been automatically added by sqlite-jdbc *\/\n  \"JDBC_EXTENSIONS\",/;" \
-	    $(SQLITE_OUT)/sqlite3.c.tmp > $(SQLITE_OUT)/sqlite3.c
-	cat src/main/ext/*.c >> $(SQLITE_OUT)/sqlite3.c
+	cp $(TARGET)/$(SQLITE_AMAL_PREFIX)/* $(SQLITE_OUT)
 	$(CC) -o $@ -c $(CCFLAGS) \
-	    -DSQLITE_ENABLE_LOAD_EXTENSION=1 \
 	    -DSQLITE_HAVE_ISNAN \
 	    -DHAVE_USLEEP=1 \
 	    -DSQLITE_ENABLE_COLUMN_METADATA \
-- 
2.34.1

