From 7a96e0c2c8dd64fb3a6abf10291eccc7bce2e65e Mon Sep 17 00:00:00 2001
From: Matthew Nelson <developer@matthewnelson.io>
Date: Sat, 30 Sep 2023 15:02:35 -0400
Subject: [PATCH 05/10] Add Linux-Musl definitions

---
 Makefile.common | 23 ++++++++++++++++++++++-
 1 file changed, 22 insertions(+), 1 deletion(-)

diff --git a/Makefile.common b/Makefile.common
index f1e260d..638c24f 100644
--- a/Makefile.common
+++ b/Makefile.common
@@ -48,7 +48,7 @@ endif
 
 # os=Default is meant to be generic unix/linux
 
-known_targets := Linux-x86 Linux-x86_64 Linux-arm Linux-armv6 Linux-armv7 Linux-Android-arm Linux-Android-aarch64 Linux-Android-x86 Linux-Android-x86_64 Linux-ppc64 Mac-x86 Mac-x86_64 Mac-aarch64 DragonFly-x86_64 FreeBSD-x86 FreeBSD-x86_64 FreeBSD-aarch64 OpenBSD-x86_64 Windows-x86 Windows-x86_64 Windows-armv7 Windows-aarch64 SunOS-sparcv9 HPUX-ia64_32
+known_targets := Linux-x86 Linux-x86_64 Linux-Musl-x86 Linux-Musl-x86_64 Linux-Musl-aarch64 Linux-arm Linux-armv6 Linux-armv7 Linux-Android-arm Linux-Android-aarch64 Linux-Android-x86 Linux-Android-x86_64 Linux-ppc64 Mac-x86 Mac-x86_64 Mac-aarch64 DragonFly-x86_64 FreeBSD-x86 FreeBSD-x86_64 FreeBSD-aarch64 OpenBSD-x86_64 Windows-x86 Windows-x86_64 Windows-armv7 Windows-aarch64 SunOS-sparcv9 HPUX-ia64_32
 target := $(OS_NAME)-$(OS_ARCH)
 
 ifeq (,$(findstring $(strip $(target)),$(known_targets)))
@@ -81,6 +81,27 @@ Linux-x86_64_LINKFLAGS := $(Default_LINKFLAGS)
 Linux-x86_64_LIBNAME   := libsqlitejdbc.so
 Linux-x86_64_SQLITE_FLAGS  := 
 
+Linux-Musl-x86_CC        := $(CROSS_PREFIX)gcc
+Linux-Musl-x86_STRIP     := $(CROSS_PREFIX)strip
+Linux-Musl-x86_CCFLAGS   := -Ilib/inc_linux -I$(JAVA_HOME)/include -Os -fPIC -m32 -fvisibility=hidden
+Linux-Musl-x86_LINKFLAGS := $(Default_LINKFLAGS)
+Linux-Musl-x86_LIBNAME   := libsqlitejdbc.so
+Linux-Musl-x86_SQLITE_FLAGS  :=
+
+Linux-Musl-x86_64_CC        := $(CROSS_PREFIX)gcc
+Linux-Musl-x86_64_STRIP     := $(CROSS_PREFIX)strip
+Linux-Musl-x86_64_CCFLAGS   := -Ilib/inc_linux -I$(JAVA_HOME)/include -Os -fPIC -m64 -fvisibility=hidden
+Linux-Musl-x86_64_LINKFLAGS := $(Default_LINKFLAGS)
+Linux-Musl-x86_64_LIBNAME   := libsqlitejdbc.so
+Linux-Musl-x86_64_SQLITE_FLAGS  :=
+
+Linux-Musl-aarch64_CC        := $(CROSS_PREFIX)gcc
+Linux-Musl-aarch64_STRIP     := $(CROSS_PREFIX)strip
+Linux-Musl-aarch64_CCFLAGS   := -I$(JAVA_HOME)/include -Ilib/inc_linux -Os -fPIC -fvisibility=hidden
+Linux-Musl-aarch64_LINKFLAGS := $(Default_LINKFLAGS)
+Linux-Musl-aarch64_LIBNAME   := libsqlitejdbc.so
+Linux-Musl-aarch64_SQLITE_FLAGS  :=
+
 Linux-arm_CC          := $(CROSS_PREFIX)gcc
 Linux-arm_STRIP       := $(CROSS_PREFIX)strip
 Linux-arm_CCFLAGS     := -I$(JAVA_HOME)/include -Ilib/inc_linux -Os -fPIC -mfloat-abi=soft -fvisibility=hidden
-- 
2.34.1

