From edc5948525ac563a92df951965ef22d9205708c6 Mon Sep 17 00:00:00 2001
From: Matthew Nelson <developer@matthewnelson.io>
Date: Thu, 16 Jan 2025 06:29:38 -0500
Subject: [PATCH 10/10] Disable cipher AEGIS

---
 Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile b/Makefile
index bb2e184..d2ec350 100644
--- a/Makefile
+++ b/Makefile
@@ -96,6 +96,7 @@ $(SQLITE_OUT)/sqlite3.o : $(SQLITE_UNPACKED)
 	    -DSQLITE_ENABLE_REGEXP=1 \
 	    -DSQLITE_TEMP_STORE=2 \
 	    -DSQLITE_USE_URI=1 \
+	    -DWXSQLITE3_HAVE_CIPHER_AEGIS=0 \
 	    $(SQLITE_FLAGS) \
 	    $(SQLITE_OUT)/sqlite3mc_amalgamation.c
 
-- 
2.34.1

