From 1545e24fa913425b01ade36371e150fd59df2364 Mon Sep 17 00:00:00 2001
From: Matthew Nelson <developer@matthewnelson.io>
Date: Sat, 16 Sep 2023 06:35:02 -0400
Subject: [PATCH 11/11] Add flag SQLITE_TEMP_STORE=2 (in memory)

---
 Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile b/Makefile
index f2c3328..602e743 100644
--- a/Makefile
+++ b/Makefile
@@ -88,6 +88,7 @@ $(SQLITE_OUT)/sqlite3.o : $(SQLITE_UNPACKED)
 	    -DCODEC_TYPE=CODEC_TYPE_CHACHA20 \
 	    -DSQLITE_ENABLE_EXTFUNC=1 \
 	    -DSQLITE_ENABLE_REGEXP=1 \
+	    -DSQLITE_TEMP_STORE=2 \
 	    -DSQLITE_USE_URI=1 \
 	    $(SQLITE_FLAGS) \
 	    $(SQLITE_OUT)/sqlite3mc_amalgamation.c
-- 
2.34.1

